### DISCUSSION TITLE
Trying to place my LinkAce in Organizer as an iFrame

### DISCUSSION THREAD

**Discussion by b52src** [Kovah/LinkAce#420] [Support] [answered]
Hello,

Im trying to place my LinkAce in Organizr as a iFrame, but i get 'X-Frame-Options' to 'SAMEORIGIN in the header  when i try to display as an iFrame. is it possible to remove this from the header so i can display it as an iFrame?

I am running both Organizer and LinkAce in Docker on a local machine currently.

**Comment by Kovah** [comment, marked as answer]
You should be able to embed LinkAce by removing `add_header X-Frame-Options "SAMEORIGIN";` from the nginx.conf file. If you are using the simple LinkAce image you would have to edit the file manually on each update.

**Reply by b52src** [reply]
thanks found it, though im seeing an issue signing in to a user while using the iframe getting an error 419 CSRF token mismatch.
is there a way to fix this?
right now im just allowing guest access to work around this.

**Reply by b52src** [reply]
never mind solved. i was going to 0.0.0.0 instead of localhost .

**Comment by SighMonIs** [comment]
I'm attempting to do the same thing. I have removed the add_header X-Frame-Options "SAMEORIGIN"; from the nginx.conf file (and confirmed I saved it properly) but I don't believe this has actually updated to nginx, I have checked with both FireFox and Chrome inspector and they both report that it's still in there. I have tried reloading the conf file (nginx -s reload) to force an update but that hasn't worked. 

I've been searching for the answer for a while now but haven't been able to get it working. Am I missing something? I am using TrueNAS Scale, which I believe uses docker for all apps, not sure if this would make a difference at all.

**Reply by Kovah** [reply]
Hi, could you tell more about the nginx.conf file you mentioned? Which exact version of Linkace are you using (PHP, Docker) and how did you change the file?

**Reply by SighMonIs** [reply]
Hi!

The nginx.conf file is located in /app/resources/docker/nginx (the only nginx.conf file I could find). I'm using the docker version (1.10.4 according to the TrueNAS apps page) I changed the file by using the unix vi command in the build in shell for TrueNAS Scale, once modified I exited, went back through shell and double checked that the file was modified.

**Reply by Kovah** [reply]
Ah okay, yeah you have to place that nginx.conf you find outside of the Docke container (somewhere on your NAS, probably) and then "put it into the container". There is an example with the exact location here: https://github.com/Kovah/LinkAce/blob/main/docker-compose.production-simple.yml#L31
Only this way Linkace itself can adopt the configuration changes.

**Reply by SighMonIs** [reply]
That makes sense. I'll have a play around and see if I can get it working, I can add the link between the folder and docker container but it won't generate the nginx.conf file. I'm still getting used to permissions, so have probably done something weird.

Thanks for your help, loving the app so far!

**Reply by SighMonIs** [reply]
I've FINALLY managed to figure it out. 
For those that may have this issue in the future here is what I did (as someone with limited knowledge of what he is doing!)

1. Downloaded the Linkace install zip file from git. Extract the "nginx.conf" file from the zip and place it where you would like the file to be on your local drive and rename it "default.conf". (I found this easier than using shell to copy and create a file as I don't know unix commands that well)
2. Install and launch Linkace in TrueNAS Scale as usual. Go into the shell for the app and navigate to /etc/nginx/http.d/
3. Open the file "default.conf" using command "vi default.conf".
4. Compare your local "default.conf" file with the shell "default.conf" file. I had to change the following lines in my local file:
        Line 11: "listen 0.0.0.0:8080" to "listen 0.0.0.0:80"
        Line 14: "add_header X-Frame-Options "SAMEORIGIN";" to "# add_header X-Frame-Options "SAMEORIGIN";"
                     you could technically delete this but I left it in just in case it's needed in the future
        Line 48: "fastcgi_pass app:9000" to "fastcgi_pass 127.0.0.1:9000"
5. Add storage, Host Path (Simple). Host path to where you have your local "default.conf" file and Mount Path to "/etc/nginx/http.d/". The last / is important as it wouldn't work for me if it wasn't there, which it usually isn't.

Now once you relaunch Linkace it should be viewable in organizr or whatever else you are using that needs an iFrame.

Thanks Kovah for your help, took me a while to figure it out but it's done!
